lib/engine/step/tracker.rb

```diff
@@ -71,14 +71,22 @@ def lay_tile(action, extra_cost: 0, entity: nil)
         discount = 0
 
         entity.abilities(:tile_lay) do |ability|
-          next if !ability.hexes.include?(hex.id) || !ability.tiles.include?(tile.name)
+          next unless ability.hexes.empty?
+
+          corporation = entity.owner
+          @game.game_error("Track laid in must be connected to one of #{corporation.id}'s stations") if
+            !@game.graph.reachable_hexes(corporation)[hex] && hex.name != corporation.coordinates
+        end if entity.company?
+
+        entity.abilities(:tile_lay) do |ability|
+          next if ability.hexes.any? && (!ability.hexes.include?(hex.id) || !ability.tiles.include?(tile.name))
 
           free = ability.free
           discount = ability.discount
         end
 
         entity.abilities(:teleport) do |ability, _|
-          ability.use! if ability.hexes.include?(hex.id) && ability.tiles.include?(tile.name)
+          ability.use! if ability.hexes.any? && ability.hexes.include?(hex.id) && ability.tiles.include?(tile.name)
```

Comment:
don't you want to use the ability regardless? if abilitiy.hexes is an empty array, you'll never use the teleport ability

Reply:
Changed backed.